sigma/plugins/fun/xkcd.py

```diff
@@ -14,22 +14,35 @@
 
     random_comic = (len(args) == 0) #True if args is empty, False otherwise
 
-    while True:
-        if not random_comic: joke_url = 'http://xkcd.com/' + comic_no + '/info.0.json'
-        else:                joke_url = 'http://xkcd.com/info.0.json'
+    if not random_comic:
+        joke_url = 'http://xkcd.com/' + comic_no + '/info.0.json'
 
         async with aiohttp.ClientSession() as session:
```

Comment:
Does this have to be inside the `random_comic` conditional? You are assigning a different url based on whether it's a random comic or not, and that's all that should be in the if/else. The rest uses the url based on whatever it is.

Reply:
Good point. Thanks for the checkup.

Fixed in b90483b